### PR TITLE
Retrieve bundle changes from bundle service when in anonymous mode.

### PR DESCRIPTION
**Pull request by frankban**
This way it's still possible to add bundles to the canvas even when the GUI is in connected to the controller but not authenticated yet.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6585/

**Comment by frankban** [comment]
QA:
- switch to production jimm;
- run `make run` on this branch;
- run `guiproxy -uuid disconnected` so that gisf mode is simulated;
- visit the proxy address and ensure you are logged out (anonymous user);
- add a bundle to the canvas: it should load the uncommitted state as usual.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6586/

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

File: jujugui/static/gui/src/app/store/env/controller-api.js [hunk 1]
```diff
@@ -387,6 +393,28 @@ YUI.add('juju-controller-api', function(Y) {
...
+        and a list of bundle changes.
+    */
+    _getBundleChangesFromBundleService: function(bundleYAML, callback) {
+      this.get('bundleService').getBundleChangesFromYAML(
+        bundleYAML, (error, changes) => {
+          if (error) {
+            callback([error], []);
+            return;
+          }
+          callback([], changes);
```
Comment (hatched) [inline jujugui/static/gui/src/app/store/env/controller-api.js]: If there are no errors then the error position in the callback should be `null`. I know that this callback expects an array but that fix should be propagated up and down.

**Pull request merged**